src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/OpenProjectsHandler.cs

```diff
@@ -17,18 +18,22 @@ internal class OpenProjectHandler : ILspServiceNotificationHandler<OpenProjectHa
 {
     private readonly LanguageServerProjectSystem _projectSystem;
 
+    private readonly IAsynchronousOperationListener _listener;
+
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public OpenProjectHandler(LanguageServerProjectSystem projectSystem)
+    public OpenProjectHandler(LanguageServerProjectSystem projectSystem, IAsynchronousOperationListenerProvider listenerProvider)
     {
         _projectSystem = projectSystem;
+        _listener = listenerProvider.GetListener(FeatureAttribute.Workspace);
     }
 
     public bool MutatesSolutionState => false;
     public bool RequiresLSPSolution => false;
 
     Task INotificationHandler<NotificationParams, RequestContext>.HandleNotificationAsync(NotificationParams request, RequestContext requestContext, CancellationToken cancellationToken)
     {
+        using var token = _listener.BeginAsyncOperation(nameof(OpenProjectHandler));
         return _projectSystem.OpenProjectsAsync(request.Projects.SelectAsArray(p => p.LocalPath));
```

Comment:
This may not be necessary in this case, given https://github.com/dotnet/roslyn/blob/5b3d150458030631797b4fb8c60549a835ea9903/src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs#L127 the method here is waiting on the result of the queue.